### ISSUE TITLE
ArchLinux ARM | Booting Stuck with Stack Trace

### ISSUE THREAD

**Issue by nar3nd3r** [utmapp/UTM#4014]
**Describe the issue**
After upgrading to UTM to latest version, I am not able to run Arch (even freshly downloaded from Gallery).

I am also sharing a crash log (not sure, if its helpful in this case).

**Configuration**
* UTM Version: Version 3.2.3 (57)
* OS Version: Apple M1 Max on 12.3.1
* Intel or Apple Silicon? Apple Silicon

**Crash log**
Attached here:
[UTM-2022-05-12-153850.log](https://github.com/utmapp/UTM/files/8691756/UTM-2022-05-12-153850.log)
[UTM-2022-05-12-190735.log](https://github.com/utmapp/UTM/files/8691757/UTM-2022-05-12-190735.log)

**Debug log**
[boot.log](https://github.com/utmapp/UTM/files/8691755/boot.log)

**Upload VM**
[Arch](https://mac.getutm.app/gallery/archlinux-arm)

**Comment by osy** [comment]
Hmm, so did UTM actually crash? If it crashed, I don't know how you're getting the boot log. The crash log indicates you've booted a VM with display emulation but the Arch VM is in console mode. I think they're not related.

Are you referring to kernel panic inside the VM as a "crash"? Because a crash refers to the UTM app terminating.

**Comment by osy** [comment]
Also can you post debug.log? The instructions should be in the template when making an issue. This will contain the QEMU command.

**Comment by nar3nd3r** [comment]
That is correct yes, It didn't crash but the app did freeze when I tried to boot it.

Also the debug log:
[debug.log](https://github.com/utmapp/UTM/files/8692041/debug.log)

**Comment by osy** [comment]
If it was frozen how did you get boot.log?

**Comment by nar3nd3r** [comment]
I copied it somehow after waiting for a while and selected (to clipboard) and pasted to text editor :P 

I know it sounds weird, the behaviour is not so certain, at times, it boots 
 and stuck on console, the other times it just sits there and the spinning wheel just rotates.

btw, Can you draw something from debug.log.

**Comment by osy** [comment]
I know there’s some kernel panic issue with M1 Max (which I don’t have to test) that was supposed to be fixed. That’s what the highmem=off was for. The other thing I can think of is to update to 12.4 and see if that makes a difference. Otherwise I’m not exactly sure what’s happening. You can confirm this doesn’t happen in 3.1.5?

**Comment by nar3nd3r** [comment]
I can confirm that the arch boot up on 3.1.5 smoothly, however after upgrading its looses it connectivity and I don't see any network except loopback.

**Comment by shlyakpavel** [comment]
```
qemu-aarch64-softmmu: -netdev vmnet-macos,mode=shared,id=net0: Parameter 'type' does not accept value 'vmnet-macos'
```
On Ubuntu here. I think they have broken the net somehow

**Comment by osy** [comment]
@shlyakpavel are you using custom args? If so delete it and use the network gui

**Comment by shlyakpavel** [comment]
@osy I'm using Ubuntu template from the website

**Comment by osy** [comment]
Post your config.plist from inside the .utm

**Comment by shlyakpavel** [comment]
[config.plist.zip](https://github.com/utmapp/UTM/files/8693017/config.plist.zip)

**Comment by shlyakpavel** [comment]
I have tried setting different network cards to solve the problem

**Comment by osy** [comment]
@shlyakpavel to be clear you’re running 3.2.3 from GitHub releases?

**Comment by shlyakpavel** [comment]
I'm running Version 3.1.5 (53) from homebrew

**Comment by osy** [comment]
@nar3nd3r When it gets stuck, can you get a spindump?

1. Open Activity Monitor
2. Find UTM process and double click
3. Click "Sample", copy paste the text to a .txt file, and attach it here.

**Comment by singh-narender** [comment]
I just upgraded to 3.2.4, yet again same issue. here is the boot log and the "Sample", though not freezing seen so far just stuck with `end trace`.

Can you help me understanding the bot log, why it failing. I can try things differently.
Thanks in advance.

[Sample of UTM.txt](https://github.com/utmapp/UTM/files/8747641/Sample.of.UTM.txt)
[boot.log](https://github.com/utmapp/UTM/files/8747642/boot.log)

**Comment by osy** [comment]
Is the UI completely frozen and you can’t click anything?

**Comment by singh-narender** [comment]
That’s what I mentioned before, no freezing this time. Just no booting.

**Comment by wjwOffline** [comment]
I also found this issue, I change cpu option to cortex-a72, it works. the default ones don't work.

**Comment by singh-narender** [comment]
thanks @wjwOffline, that was awesome. Any tip on the following scenario, updating with `pacman -Syyu` and reboot leads to missing lan afterwards.

**Comment by shlyakpavel** [comment]
> Same here, no network after updating and rebooting. Any solutions?

Of course there is a solution. Kernel modules have been updated, while kernel itself hasn't. Just take Image.gz and initramfs-linux.img from the system i9mage and replace them in ArchLinux ARM.utm/images

**Comment by shlyakpavel** [comment]
> Same here, no network after updating and rebooting. Any solutions?

Check https://github.com/utmapp/UTM/issues/3888

**Comment by mkurz** [comment]
> Just take Image.gz and initramfs-linux.img from the system i9mage and replace them in ArchLinux ARM.utm/images

Can you be a bit more specific? Take what from where exactly and put it where exactly? Thanks!

**Comment by shlyakpavel** [comment]
> Can you be a bit more specific? 

Just right click "ArchLinux ARM.utm", "Open package", go to "images" directory and replace files with files extracted from this zip. Never update the kernel again.
[utm.zip](https://github.com/utmapp/UTM/files/8770579/utm.zip)

**Comment by shlyakpavel** [comment]
@mkurz 
Also, ensure these entries still point to the updated files
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/19782228/170251919-ac556bce-8f17-4270-b640-783afff8f012.png">
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/19782228/170251958-e3551529-6777-4a92-97a3-34debdff989b.png">

**Comment by mkurz** [comment]
Thanks!

**Comment by mkurz** [comment]
BTW: Did you install plasma? If so do you use it with wayland? Are your fonts rendered ok?

**Comment by shlyakpavel** [comment]
> BTW: Did you install plasma? If so do you use it with wayland? Are your fonts rendered ok?

I did not install GUI. I use it as a build machine via ssh.

**Comment by h0m3** [comment]
> Thanks @shlyakpavel I was able to create a base with Gnome installed on top of it. works okay, graphics looks a bit odd, but will see.
> 
> ![Screen Shot 2022-05-25 at 1 21 28 PM](https://user-images.githubusercontent.com/39687188/170325148-75c79fa0-4328-4f8f-a2cd-90f4ca9c43ef.png)
> 
> For any one coming this far, please follow installation from scratch as mentioned here in the [Blog of ktprograms](https://ktprograms.codeberg.page/blog/posts/2022-03-17_1750_utm-arch-arm/)

Had the same issue, following the guide worked for me.

**Comment by StathisKap** [comment]
> I also found this issue, I change cpu option to cortex-a72, it works. the default ones don't work.

On my 2021 14inch M1 Pro, cortex-a72 didn't work. cortex-a53 did.

**Comment by DemonJun** [comment]
on my 2021 14inch M1 MAX, cpu option to host work
